torch/utils/data/dataloader.py

```diff
@@ -219,6 +219,9 @@ def __init__(self, loader):
             # prime the prefetch loop
             for _ in range(2 * self.num_workers):
                 self._put_indices()
+        elif self.num_workers < 0:
```

Comment:
Actually now that I think about it, this check should probably happen when constructing the data loader and not the iterator. Can you please move it there?